File: hivepulse-api/internal/domain/stats.go
package domain

import "time"

type StatsBucket struct {
	Time       time.Time `json:"time"`
	UpCount    int       `json:"up_count"`
	TotalCount int       `json:"total_count"`
	AvgPingMS  int       `json:"avg_ping_ms"`
}

type DownPeriod struct {
	StartedAt  time.Time  `json:"started_at"`
	ResolvedAt *time.Time `json:"resolved_at,omitempty"`
}

type StatsResponse struct {
	UptimePct   float64        `json:"uptime_pct"`
	AvgPingMS   int            `json:"avg_ping_ms"`
	Buckets     []*StatsBucket `json:"buckets"`
	DownPeriods []*DownPeriod  `json:"down_periods"`
}

File: hivepulse-web/src/domain/stats.ts
export interface StatsBucket {
  time: string
  up_count: number
  total_count: number
  avg_ping_ms: number
}

export interface DownPeriod {
  started_at: string
  resolved_at?: string // omitted (not null) for active incidents
}

export interface StatsResponse {
  uptime_pct: number
  avg_ping_ms: number
  buckets: StatsBucket[]
  down_periods: DownPeriod[]
}

export type StatsRange = '1h' | '3h' | '6h' | '24h' | '48h' | '7d' | '15d' | '30d' | '90d'

export interface OverviewStats {
  avg_ping_ms: number
  buckets: StatsBucket[]
}
